### PR TITLE
Build OpenMM 7.7.0

### PR DESCRIPTION
**Pull request by peastman**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
The Windows Python 3.9 builds are failing with this error:

```
Traceback (most recent call last):
  File "D:\bld\openmm_1640119062043\test_tmp\run_test.py", line 2, in <module>
    import openmm
  File "D:\bld\openmm_1640119062043\_test_env\lib\site-packages\openmm\__init__.py", line 19, in <module>
    from openmm.openmm import *
  File "D:\bld\openmm_1640119062043\_test_env\lib\site-packages\openmm\openmm.py", line 13, in <module>
    from . import _openmm
ImportError: DLL load failed while importing _openmm: The specified module could not be found.
```

They worked just a week ago when I built the release candidate, and no code changes have happened since then.  Does anyone know what is causing this?

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
I tried making the change suggested in https://github.com/openmm/openmm/pull/3390#issuecomment-999775752, but that causes all the builds to fail.  Any suggestions for the correct way to fix it?

**Comment by jaimergp** [comment]
The Python version is controlled through the `.ci_config/*` files, so my first try would be modifying  the Windows 3.9 YAMLs. Or maybe with some Jinja in `meta.yaml` and `run_constrained`?

**Comment by peastman** [comment]
Do you mean put the `python <3.9.9 # [win]` into the `run_constrained` section?

**Comment by jaimergp** [comment]
In a way. The thing is that that constrain should only apply to the py39 branch. Rerendering with those constrains on host/run removed all the python matrix entries. You need to undo that change and try with the `run_constrained` trick but within a Jinja if check against the python version. Something like:

```yaml
run_constrained:
{% if py == 39 %}
run_constrained:
  - python <=3.9.9
{% endif %}
```

I don't remember the syntax well but that's the idea.

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
All builds passed!  Shall I merge it?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -76,6 +76,9 @@ requirements:
   run_constrained:
     - khronos-opencl-icd-loader  ==9999999999  # [osx and opencl_impl == 'apple']
     - ocl_icd_wrapper_apple      ==9999999999  # [osx and opencl_impl == 'apple']
+{% if py == 39 %}
+    - python <3.9.9  # [win]
+{% endif %}
```
Comment (isuruf) [inline recipe/meta.yaml:81]: This should be a run requirement instead of `run_constrained`.

```suggestion
    - python <3.9.9  # [py == 39]
```

Comment (peastman) [reply recipe/meta.yaml:81]: Wouldn't that affect all platforms rather than just Windows?

Comment (isuruf) [reply recipe/meta.yaml:81]: Ah yes. `python <3.9.9  # [py == 39 and win]`

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -76,6 +76,7 @@ requirements:
   run_constrained:
     - khronos-opencl-icd-loader  ==9999999999  # [osx and opencl_impl == 'apple']
     - ocl_icd_wrapper_apple      ==9999999999  # [osx and opencl_impl == 'apple']
+    - python <3.9.9  # [py == 39 and win]
```
Comment (isuruf) [inline recipe/meta.yaml:79]: Can you move this to `run`?

Comment (peastman) [reply recipe/meta.yaml:79]: Is that in addition to listing just `python` under `run`, or instead of it?

Comment (isuruf) [reply recipe/meta.yaml:79]: In addition

Comment (peastman) [reply recipe/meta.yaml]: That didn't work.  It still installed 3.9.9.

**Comment by peastman** [comment]
All builds are passing, so I'm going to merge this.  Then we need to add the same constraint to older releases to get them working on Windows again.  And then we can finish debugging https://github.com/openmm/openmm/pull/3390 which is the long term fix, but there's no rush about that.  Thanks everyone!

**Comment by peastman** [comment]
I don't see any sign of it rebuilding.  Do I need to do something else to force it to build?

**Comment by jaimergp** [comment]
Hm, it wasn't triggered. Try sending an empty commit to master.

**Comment by peastman** [comment]
Thanks!  That seems to be working.

**Pull request merged**